lib/active_fedora/file_persistence.rb

```diff
@@ -0,0 +1,27 @@
+module ActiveFedora
+  module FilePersistence
+    extend ActiveSupport::Concern
+
+    include ActiveFedora::Persistence
+
+    private
+
+      def _create_record(_options = {})
+        return false if content.nil?
+        ldp_source.content = content
+        ldp_source.create do |req|
+          req.headers.merge!(ldp_headers)
+        end
+        refresh
+      end
+
+      def _update_record(_options = {})
+        return false unless content_changed?
```

Comment:
I looked at how Rails handles callbacks -- they don't suppress the `after_update` hook if nothing changed (and, presumably, neither do we, in `AF::Persistence`). I've updated this behavior in 5bae75f.